lib/ansible/plugins/callback/logdna.py

```diff
@@ -0,0 +1,204 @@
+        return IP
+
+
+# Is it JSON?
+def isJSONable(obj):
+    try:
+        json.dumps(obj)
+        return True
+    except BaseException:
+        return False
+
+
+# LogDNA Callback Module:
+class CallbackModule(CallbackBase):
+
+    CALLBACK_VERSION = 0.1
+    CALLBACK_TYPE = 'aggregate'
+    CALLBACK_NAME = 'logdna'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+        super(CallbackModule, self).__init__(display=display)
+
+        self.disabled = True
+        self.playbook_name = None
+        self.playbook = None
+        self.conf_key = None
+        self.plugin_ignore_errors = None
+        self.conf_hostname = None
+        self.conf_tags = None
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.conf_key = self.get_option('conf_key')
+        self.plugin_ignore_errors = self.get_option('plugin_ignore_errors')
+        self.conf_hostname = self.get_option('conf_hostname')
+        self.conf_tags = self.get_option('conf_tags')
+        self.mac = get_mac()
+        self.ip = get_ip()
+
+        if self.plugin_ignore_errors is None:
+            self.plugin_ignore_errors = False
+
+        if self.conf_hostname is None:
+            self.conf_hostname = get_hostname()
+
+        if self.conf_tags is None:
+            self.conf_tags = ['ansible']
+        else:
+            if isinstance(self.conf_tags, str):
+                self.conf_tags = self.conf_tags.split(',')
+            elif not isinstance(self.conf_tags, list):
+                self.conf_tags = [str(self.conf_tags)]
+
+        if self.conf_key is None:
+            self.disabled = True
+            self._display.warning('WARNING:\nLogDNA Ingestion Key has not been provided!')
+        else:
+            self.disabled = False
+            if HAS_LOGDNA:
+                self.log = logging.getLogger('logdna')
+                self.log.setLevel(logging.INFO)
+                self.options = {'hostname': self.conf_hostname, 'mac': self.mac, 'index_meta': True}
+                self.log.addHandler(LogDNAHandler(self.conf_key, self.options))
+                self.disabled = False
+            else:
+                self.disabled = True
+                self._display.warning('WARNING:\nPlease, install LogDNA Python Package: `pip install logdna`')
+
+    def metaIndexing(self, meta):
+        invalidKeys = []
+        ninvalidKeys = 0
+        for key, value in meta.items():
+            if not isJSONable(value):
+                invalidKeys.append(key)
+                ninvalidKeys += 1
+        if ninvalidKeys > 0:
+            for key in invalidKeys:
+                del meta[key]
+            meta['__errors'] = 'These keys have been sanitized: ' + ', '.join(invalidKeys)
+        return meta
+
+    def sanitizeJSON(self, data):
+        try:
+            return json.loads(json.dumps(data))
```

Comment:
i recommend using the ansible json encoder as it deals with 'special' types and will avoid a lot of serialization errors you'll get with the stock json.

Reply:
@bcoca, I couldn't find out how to use ansible json encoder in ansible callback plugin development - can you give me an example and/or can you make a commit to this Pull Request? All of your change requests except this are ready.

Reply:
Used by the base class to be able to handle all ansible data https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/callback/__init__.py#L123